sys/psa_crypto/psa_key_slot_mgmt/psa_crypto_slot_management.c

```diff
@@ -287,37 +276,110 @@ static psa_status_t psa_get_and_lock_key_slot_in_memory(psa_key_id_t id, psa_key
 {
     psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
 
-    if (psa_key_id_is_volatile(id)) {
-        clist_node_t *slot_node = clist_foreach(&key_slot_list, node_id_equals_key_id, &id);
-        if (slot_node == NULL) {
-            return PSA_ERROR_INVALID_HANDLE;
-        }
+    clist_node_t *slot_node = clist_foreach(&key_slot_list, node_id_equals_key_id, &id);
+    if (slot_node == NULL) {
+        return PSA_ERROR_DOES_NOT_EXIST;
+    }
 
-        psa_key_slot_t *slot = container_of(slot_node, psa_key_slot_t, node);
-        status = psa_lock_key_slot(slot);
-        if (status == PSA_SUCCESS) {
-            *p_slot = slot;
-        }
+    psa_key_slot_t *slot = container_of(slot_node, psa_key_slot_t, node);
+    status = psa_lock_key_slot(slot);
+    if (status == PSA_SUCCESS) {
+        *p_slot = slot;
+    }
+    return status;
+}
+
+#if IS_USED(MODULE_PSA_PERSISTENT_STORAGE)
+/**
+ * @brief   Reads a persistently stored key from storage
+ *
+ * @param   id        ID of the desired key
+ * @param   p_slot    Pointer to store key slot in
+ * @return  psa_status_t
+ */
+static psa_status_t psa_get_persisted_key_slot_from_storage(psa_key_id_t id,
+                                                            psa_key_slot_t **p_slot)
+{
+    uint8_t cbor_buf[PSA_MAX_KEY_SLOT_SIZE*2];
```

Comment:
Hmm, this is a "just in case" thing, since the size of the cbor output with  encoding will be larger than the key slot and can slightly vary in size, depending on the individual values.  
But yeah, I also don't like this very much. I'll think about it.

Reply:
I changed this to only `PSA_MAX_KEY_SLOT_SIZE`. Since I'm using bytestrings for keys now, the encoding size will not vary as much, so that should be enough.

Reply:
Hum, I still don't like this, there should be a way of deriving the actual max size by considering the maximum representations for the integers you are storing in CBOR format. If I'm not mistaken that `1+sizeof(uint*_t)`. Additionally, we have 1 byte per array marker and the `bstr`s are fixed as per the size in the respective struct.

This could still fail in case a key was persisted with a different compile-time configuration, but if I'm not mistaken this would just lead to a decoding error (when reading the file content) and is thus fine. But we should avoid failing to store/load a key with a valid configuration.

Maybe that's something worth an additional test (application): try to break the current max size and then fixing it.

Reply:
I haven't fixed this, yet, but I'm going to work on this (just not today)

Reply:
I have added an alternative. This should be more accurate.

Reply:
LGTM, thanks!